stuff/pipelines/test_course_callback.py

```diff
@@ -0,0 +1,167 @@
+        finish_time=time.mktime(future_time.timetuple()),
+        objective='Testing',
+        level='intermediate',
+        show_scoreboard=1,
+        requests_user_information='no',
+    )
+
+    client.course.createAssignment(
+        name=assignment_alias,
+        alias=assignment_alias,
+        course_alias=course_alias,
+        assignment_type='homework',
+        description='Test course assignment',
+        start_time=time.mktime(current_time.timetuple()),
+        finish_time=time.mktime(future_time.timetuple()),
+    )
+
+    assignmentDetails = client.course.assignmentDetails(
+        course=course_alias,
+        assignment=assignment_alias)
+
+    client.course.addProblem(
+        course_alias=course_alias,
+        assignment_alias=assignment_alias,
+        problem_alias='sumas',
+        points=1.0,
+    )
+
+    usernames: List[str] = []
+    for number in range(3):
+        user = f'course_test_user_{number}'
+        client.course.addStudent(course_alias=course_alias,
+                                 usernameOrEmail=user,
+                                 share_user_information=True,
+                                 accept_teacher_git_object_id='0',
+                                 privacy_git_object_id='0',
+                                 statement_type='accept_teacher',
+                                 accept_teacher=False)
+        client.run.create(
+            problemset_id=assignmentDetails.problemset_id,
+            contest_alias='', # Fix the API in order to make optional this param
+            problem_alias='sumas',
+            language='py3',
+            source='print(1)',
+        )
+        usernames.append(user)
+
+    dbconn = lib.db.connect(
+        lib.db.DatabaseConnectionArguments(
+            user=test_credentials.MYSQL_USER,
+            password=test_credentials.MYSQL_PASSWORD,
+            host=test_credentials.MYSQL_HOST,
+            database=test_credentials.MYSQL_DATABASE,
+            port=test_credentials.MYSQL_PORT,
+            mysql_config_file=lib.db.default_config_file_path() or '',
+        )
+    )
+
+    with dbconn.cursor(buffered=True, dictionary=True) as cur:
+        cur.execute(
+            '''
+            SELECT
+                c.course_id
+            FROM
+                Courses c
+            WHERE
+                alias = %s;
+            ''', (course_alias,))
+        result = cur.fetchone()
+
+    course_id = result['course_id']
+    with rabbitmq_connection.connect(
+            username=test_credentials.OMEGAUP_USERNAME,
+            password=test_credentials.OMEGAUP_PASSWORD,
+            host=test_credentials.RABBITMQ_HOST,
+    ) as channel:
+        callback = course_callback.CourseCallback(
+            dbconn=dbconn.conn,
+            client=client,
+        )
+        body = course_callback.CourseCertificate(
+            course_id=course_id,
+            minimum_progress_for_certificate=50, # mocking a default value
```

Comment:
```suggestion
            minimum_progress_for_certificate=50,
```

esto no es un mock. un [mock](https://en.wikipedia.org/wiki/Mock_object) es un objeto que simula ser otro objeto con implementación real.

Reply:
Listo